### ISSUE TITLE
kv450_cf likelihood not working

### ISSUE THREAD

**Issue by shan1525** [brinckmann/montepython_public#253]
Dear developers,

I am trying to run the kids-viking likelihood. I downloaded the likelihood file from git and saved the data and ini file  in the likelihood folder of kv450_cf folder. I added the path to cosmic-shear-release in the data file. But when I am trying to run it I am getting the following error. Please have a look and suggest what could be the problem.

Running Monte Python v3.5.0

with CLASS v3.0.1

Testing likelihoods for:
 ->kv450_cf_likelihood_public, sh0es

Using hmcode to obtain the non-linear corrections for the matter power spectrum, P(k, z)! 

Traceback (most recent call last):
  File "/mnt/home/associate/csomasri/shaz/cluster_monte/montepython_public-3.5/montepython/MontePython.py", line 40, in <module>
    sys.exit(run())
  File "/mnt/home/associate/csomasri/shaz/cluster_monte/montepython_public-3.5/montepython/run.py", line 32, in run
    custom_command)
  File "/mnt/home/associate/csomasri/shaz/cluster_monte/montepython_public-3.5/montepython/run.py", line 191, in safe_initialisation
    cosmo, data, command_line, success = initialise(custom_command)
  File "/mnt/home/associate/csomasri/shaz/cluster_monte/montepython_public-3.5/montepython/initialise.py", line 67, in initialise
    data = Data(command_line, path)
  File "/mnt/home/associate/csomasri/shaz/cluster_monte/montepython_public-3.5/montepython/data.py", line 361, in __init__
    self.initialise_likelihoods(self.experiments)
  File "/mnt/home/associate/csomasri/shaz/cluster_monte/montepython_public-3.5/montepython/data.py", line 498, in initialise_likelihoods
    elem, elem, folder, elem))
  File "<string>", line 1, in <module>
  File "/mnt/home/associate/csomasri/shaz/cluster_monte/montepython_public-3.5/montepython/likelihoods/kv450_cf_likelihood_public/__init__.py", line 125, in __init__
    temp = self.__load_public_data_vector()
  File "/mnt/home/associate/csomasri/shaz/cluster_monte/montepython_public-3.5/montepython/likelihoods/kv450_cf_likelihood_public/__init__.py", line 446, in __load_public_data_vector
    data_xip = np.zeros((self.ntheta, self.nzcorrs + 1))
TypeError: 'float' object cannot be interpreted as an integer.

I tried to search it in the issues page. But I didn't see any relevant issue. Please reply if you have any suggestions.

With the best regards,
shan.

**Comment by brinckmann** [comment]
Hi Shan,

From the error it looks like one of the variables going into the following line isn't an integer and they have to be.
`data_xip = np.zeros((self.ntheta, self.nzcorrs + 1))`

On line 76 of montepython/likelihoods/kv450_cf_likelihood_public/__init__.py we have
`self.ntheta = 9`
So that should be fine

On line 117-118 we have
```
self.nzbins = len(self.z_bins_min)
self.nzcorrs = self.nzbins * (self.nzbins + 1) // 2
```
I'm not sure how this doesn't return an integer. By definition len returns the length of an array as an integer, while in Python 3 double slash does integer division (since slash does float division). In Python 2 both // and / should result in integer division.

Are you perhaps using an old version of the likelihood with only a single slash along with Python 3? I could see how this could result in the error.

So in short, look at line 118 of montepython/likelihoods/kv450_cf_likelihood_public/__init__.py and make sure it has a double slash in front of the divisor. If this doesn't work look at line 446 and print the variables in question just before it to see what the values are, maybe it'll give us an idea of what is going on.

Best,
Thejs

**Comment by shan1525** [comment]
Dear @brinckmann , 
Thank you so much for your prompt reply. I will follow your instructions and see what happens. I will reply here what I get following your instructions.

with the best regards,
Shan.

**Comment by shan1525** [comment]
Dear @brinckmann  ,

With your help I could solve the problem. The problem was , I went through the readme.md file in the likelihood folder of kv450_cv and downloaded the ini and kv450_____.dat file from the gitclone , but I didn't realize those files were already there in montepython-3.5. So, I removed the git downloaded file and replaced it with the earlier one's. Now, The program is running. Thank you so much for making this beautiful code and always being active in the issue page. I am closing the issue now as it has been resolved.

with the best regards,
shan.
